### PR TITLE
Hotfix - allow class constants with all possible scalar values (PHP 5.6 compatible)

### PR DESCRIPTION
**Pull request by Ocramius**

### CODE REVIEW

**Comment by asgrim** [comment]
:+1:

**Comment by Ocramius** [comment]
Merging myself after review by @asgrim.

Feel free to punch me afterwards

**Comment by Ocramius** [comment]
Merged:

`master`: zendframework/zend-code@bd476716bce6edcfeb46e4f2c50936c5ddd443f9
`develop`: zendframework/zend-code@1cf4d81a329b2f4a92ac25065f40bd8fa5bdaeb3

File: test/Generator/ClassGeneratorTest.php [hunk 1]
```diff
@@ -563,16 +564,69 @@ public function testAddConstantThrowsExceptionWithInvalidName()
...
+
+        $this->setExpectedException(InvalidArgumentException::class);
+
+        $classGenerator->addConstant('', 'value');
+    }
+
+    public function testAddConstantAcceptsMixedScalars()
+    {
+        $classGenerator = new ClassGenerator();
+
```
Comment (Maks3w) [inline test/Generator/ClassGeneratorTest.php]: This should be a data provider

Comment (Maks3w) [inline test/Generator/ClassGeneratorTest.php]: I would like examples for arithmetic expressions. 

2 + - / * ** % 2  + Another::COSTANT

File: test/Generator/ClassGeneratorTest.php [hunk 2]
```diff
@@ -563,16 +564,69 @@ public function testAddConstantThrowsExceptionWithInvalidName()
...
-     */
-    public function testAddConstantThrowsExceptionWithInvalidValue()
+    public function testAddConstantThrowsExceptionWithEmptyConstantName()
     {
-        $this->setExpectedException('InvalidArgumentException');
+        $classGenerator = new ClassGenerator();
+
+        $this->setExpectedException(InvalidArgumentException::class);
+
+        $classGenerator->addConstant('', 'value');
```
Comment (Maks3w) [inline test/Generator/ClassGeneratorTest.php]: Same thing about use data providers

**Pull request merged**